File: src/BackEnd/SnowyDune/src/main/java/org/iesalixar/poriad/entity/CarRental.java
package org.iesalixar.poriad.entity;

import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class CarRental {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private Double price;
	
	private String description;
	
	private String name;
	
	private String phone;
	
	private String email;
	
	private String location;
	
	private String urlImages;
	
	private Integer activated;
	
	@ManyToOne
	@JoinColumn(name = "station_id", nullable = true)
	private Station station;
	
	@JsonIgnore
	@OneToMany(mappedBy = "carRental", fetch = FetchType.LAZY)
	private Set<Comment> comments;
	
	@JsonIgnore
	@OneToMany(mappedBy = "carRental")
	private Set<Trip> trip;
	
}

File: src/FrontEnd/snowy-dune/src/app/models/car-rental.ts
export class CarRental {
    id?: number;
    price: number;
    description: string;
    name: string;
    phone: string;
    email: string;
    location: string;
    urlImages: string;
    activated: number;

    constructor(id: number, price: number, name: string, phone: string,email: string, location: string, urlImages: string, activated: number){
        this.id = id;
        this.price = price;
        this.name = name;
        this.phone = phone;
        this.email = email;
        this.location = location;
        this.urlImages = urlImages;
        this.activated = activated;
    }
}
